tidal/content/contents/code/tidal.js

```diff
@@ -132,142 +106,192 @@ var TidalResolver = Tomahawk.extend( TomahawkResolver, {
         return entry.name;
     },
 
-    search: function (qid, query) {
-        if (!this._countryCode) return;
+    search: function (query) {
+        if (!this.logged_in) {
+            this._login(this.search, [query], this);
+            return;
+        }
+
         var that = this;
+        var query = encodeURIComponent(query);
+        var params = {
+            limit: 25,
+            query: query,
+            token: this.api_token,
+            countryCode: this._countryCode
+        };
 
-        query = encodeURIComponent(query);
+        tracks = Tomahawk.get(this.api_location + "search/tracks", {
+                data: params
+            }).then( function (response) {
+                return response.items.map(that._convertTrack, that);
+        });
 
-        xhrRequest(that.api_location + "search/tracks","GET", "limit=10000&query=" 
-                +query+"&token="+that.api_token+"&countryCode="+that._countryCode,
-            function (request) {
-                var resp = JSON.parse(request.responseText);
-                var data = {
-                    qid: qid,
-                    results: resp.items.map(that._convertTrack),
-                }
-                Tomahawk.addTrackResults(data);
+        return tracks;
+
+        /*
+        albums = Tomahawk.get(this.api_location + "search/albums", {
+                data: params
+            }).then( function (response) {
+                var result =  response.items.map(that._convertAlbum, that);
+                return result;
+        });
+
+        artists = Tomahawk.get(this.api_location + "search/artists", {
+                data: params
+            }).then( function (response) {
+                return response.items.map(that._convertArtist, that);
+        });
+
+        // WIP, many problems with this. Promise.all doesn't seem to work.
```

Comment:
Well, it never got forefilled. Despite the previous three all forefilling fine. But doesn't really matter anyway till Tomahawk wants it.